### PR TITLE
feat: add grpc static library

### PR DESCRIPTION
**Pull request by Nateiru**
1. add grpc static library as submoduel 
2. use the grpc in submodule
3. add how to use greptimedb-client-cpp in README.md

### CODE REVIEW

**Comment by v0y4g3r** [comment]
Can you add some instructions on how to compile greptimedb-client-cpp with dependencies managed by git submodules?

**Comment by Nateiru** [comment]
> Can you add some instructions on how to compile greptimedb-client-cpp with dependencies managed by git submodules?

**Comment by Nateiru** [comment]
> Can you add some instructions on how to compile greptimedb-client-cpp with dependencies managed by git submodules?

OK, I have added how to use greptimedb-client-cpp in README.md

File: README.md [hunk 1]
```diff
@@ -11,6 +11,27 @@
...
 
 ```bash
-git submodule update --init --recursive 
-cmake --build .
+git clone git@github.com:GreptimeTeam/greptimedb-client-cpp.git
+
+cd greptimedb-client-cpp
+
+# update submodule 
+git submodule update --init --recursive
```
Comment (v0y4g3r) [inline README.md]: Use shalow clone: 
```suggestion
git clone --recurse-submodules --depth 1 --shallow-submodules https://github.com/GreptimeTeam/greptimedb-client-cpp

cd greptimedb-client-cpp
```

**Pull request merged**